### PR TITLE
Use Token from env even github app is used

### PR DESCRIPTION
**Pull request by Jeffwan**
This is not a solid PR to address https://github.com/kubernetes/test-infra/issues/24141. 

I am not that familiar with Github App Auth, I file this PR as a workaround to keep using Github Access Token for GithubApp setup. Otherwise, seem some clients miss Auth Header and won't get fields like `allow_merge_commit`, `delete_branch_on_merge` etc. 

The detail analysis is in  https://github.com/kubernetes/test-infra/issues/24141#issuecomment-1016042616

### CODE REVIEW

File: prow/github/client.go [hunk 1]
```diff
@@ -687,8 +688,15 @@ func NewClientFromOptions(fields logrus.Fields, options ClientOptions) (TokenGen
 	options = options.Default()
 
 	// Will be nil if github app authentication is used
+	// Hack: We manually retrieve personal access token to support some operation missing GitHubApp jwtToken.
+	// This helps avoid situation that tide calls github api anonymously and leads to fields missing
 	if options.GetToken == nil {
-		options.GetToken = func() []byte { return nil }
+		personalAccessToken := os.Getenv("GITHUB_TOKEN")
```
Comment (alvaroaleman) [inline prow/github/client.go:694]: /hold

We should not mix apps aush and PAT auth. There might be a bug in the area, but this is not a reasonable solution.

Comment (Jeffwan) [reply prow/github/client.go:694]: I agree. Is there a better way that I can help with from your perspective?

Comment (Jeffwan) [reply prow/github/client.go:694]: Technically, I feel app temp token should be generated and renewed, so we can return that credential in `GetToken`. I probably need more time to dig into the code and get familiar with app authentication

Comment (alvaroaleman) [reply prow/github/client.go:694]: As mentioned in https://github.com/kubernetes/test-infra/issues/24141#issuecomment-1016624541, the issue is likely that your app has insufficient permissions, it needs `Contents: Read & Write`. Apps auth definitely does get an Authorization Header set, it just happens on a different layer in a http transport, as it needs to inspect the request in order to decide what token to set.

**Comment by Jeffwan** [comment]
As we talked in https://github.com/kubernetes/test-infra/issues/24141#issuecomment-1016624541, this PR should be closed

**Pull request closed without merging**